census.go

```diff
@@ -468,6 +468,147 @@ func (v *vocdoniHandler) censusFollowers(msg *apirest.APIdata, ctx *httprouter.H
+		// to the queue and return it to the client
+		v.censusCreationMap.Store(censusID.String(), CensusInfo{})
+		// run a goroutine to create the census, update the queue with the progress,
+		// and update the queue result when it's ready
+		go func(channelID string) {
+			internalCtx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			var err error
+			// get the fids of the users in the channel from neynar farcaster API, if
+			// the channel does not exist, return a NotFound error
+			var users []uint64
+			v.trackStepProgress(censusID, 1, 3, func(progress chan int) {
+				users, err = v.fcapi.ChannelFIDs(internalCtx, channelID, progress)
+			})
+			if err != nil {
+				log.Errorw(err, "failed to get channel fids from farcaster API")
+				v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: err.Error()})
+				return
+			}
+			if len(users) == 0 {
+				log.Errorw(fmt.Errorf("no valid participants found for the channel %s", channelID), "")
+				v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: "no valid participants found for the channel"})
+				return
+			}
+			// create the participants from the database users using the fids
+			var participants []*FarcasterParticipant
+			v.trackStepProgress(censusID, 2, 3, func(progress chan int) {
+				participants = v.farcasterCensusFromFids(users, progress)
+			})
+			if len(participants) == 0 {
+				log.Errorw(fmt.Errorf("no valid participant signers found for the channel %s", channelID), "")
+				v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: "no valid participant signers found for the channel"})
+				return
+			}
+			// create the census from the participants
+			var censusInfo *CensusInfo
+			v.trackStepProgress(censusID, 3, 3, func(progress chan int) {
+				censusInfo, err = CreateCensus(v.cli, participants, FrameCensusTypeChannelGated, 0, progress)
+			})
+			if err != nil {
+				v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: err.Error()})
+				return
+			}
+			uniqueUsernames := map[string]bool{}
+			for _, p := range participants {
+				if _, ok := uniqueUsernames[p.Username]; !ok {
+					uniqueUsernames[p.Username] = true
+				}
+			}
+			for username := range uniqueUsernames {
+				censusInfo.Usernames = append(censusInfo.Usernames, username)
+			}
+			censusInfo.FromTotalAddresses = uint32(len(participants))
+			v.censusCreationMap.Store(censusID.String(), *censusInfo)
+			log.Infow("census created from channel",
+				"channelID", channelID,
+				"participants", len(censusInfo.Usernames))
+		}(community.Census.Channel)
+		// return the censusID to the client
+		data, err := json.Marshal(map[string]string{"censusId": censusID.String()})
+		if err != nil {
+			return err
+		}
+		return ctx.Send(data, http.StatusOK)
+	}
+	// if the census type is not a channel, the type is NFT or ERC20, so create
+	// the census sync and from the token holders from airstack
+
+	// extract userFID from auth token
```

Comment:
This should be done at the beginning of the method and we should check if the user calling the endpoint is actually an admin for the community.